File: src/main/java/org/radarbase/management/service/dto/SiteSettingsDTO.java
package org.radarbase.management.service.dto;


import org.radarbase.management.config.ManagementPortalProperties;

import java.io.Serializable;
import java.util.*;

/**
 * A DTO for the {@link ManagementPortalProperties.SiteSettings} entity.
 */
public class SiteSettingsDTO implements Serializable {

    private List<String> hiddenSubjectFields = List.of();

    public List<String> getHiddenSubjectFields() {
        return hiddenSubjectFields;
    }

    public void setHiddenSubjectFields(List<String> hiddenSubjectFields) {
        this.hiddenSubjectFields = hiddenSubjectFields;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        SiteSettingsDTO that = (SiteSettingsDTO) o;
        return Objects.equals(hiddenSubjectFields, that.hiddenSubjectFields);
    }

    @Override
    public int hashCode() {
        return Objects.hash(hiddenSubjectFields);
    }

    @Override
    public String toString() {
        return "SiteSettingsDTO{" +
                "hiddenSubjectProperties=" + hiddenSubjectFields +
                '}';
    }
}

File: src/main/java/org/radarbase/management/service/dto/SiteSettingsDto.kt
package org.radarbase.management.service.dto

import org.radarbase.management.config.ManagementPortalProperties
import java.io.Serializable
import java.util.*

/**
 * A DTO for the [ManagementPortalProperties.SiteSettings] entity.
 */
class SiteSettingsDto : Serializable {
    var hiddenSubjectFields = listOf<String>()
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as SiteSettingsDto
        return hiddenSubjectFields == that.hiddenSubjectFields
    }

    override fun hashCode(): Int {
        return Objects.hash(hiddenSubjectFields)
    }

    override fun toString(): String {
        return ("SiteSettingsDTO{"
                + "hiddenSubjectProperties="
                + hiddenSubjectFields
                + '}')
    }
}
